aten/src/ATen/MemoryFormatUtils.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <ATen/ATen.h>
+
+namespace at {
+
+static Tensor clone_if_possible_with_memory_format(const Tensor& self) {
```

Comment:
I'm not sure how you're planning on using this function, but the naming of this is a bit of a mouthful